Katana/Store.swift

```diff
@@ -15,36 +15,54 @@ public protocol AnyStore: class {
   var anyState: State { get }
 
   /**
-   Dispatches a `ReturningSideEffect` item
+   Dispatches a generic `Dispatchable` item. This is useful for customizing Katana's dispatchable, for example in other libraries.
 
-   - parameter dispatchable: the Returning Side Effect to dispatch
-   - returns: a promise parameterized to the side effect's return value, that is resolved when the dispatchable is handled by the store
+   - parameter dispatchable: the item to dispatch
+   - returns: a promise that is resolved when the dispatchable is handled by the store, resolving to a value associated with the dispatchable
    */
   @discardableResult
-  func dispatch<T: ReturningSideEffect>(_ dispatchable: T) -> Promise<T.ReturnValue>
-  
+  func anyDispatch(_ dispatchable: Dispatchable) -> Promise<Any>
+
+  /**
+   Dispatches a `Dispatchable` item. This is useful for customizing Katana's dispatchable, for example in other libraries.
+
+   - parameter dispatchable: the item to dispatch
+   - returns: a promise that is resolved when the dispatchable is handled by the store, resolving to a value associated with the dispatchable
+   */
+  @discardableResult
+  func anyDispatch<T: Dispatchable>(_ dispatchable: T) -> Promise<Any>
```

Comment:
I feel like this is unneeded, since we already have a non-generics version which should cover all the usecases